### PR TITLE
feat: add paseto extension

### LINKED ISSUE #618: Proposal: add paseto middleware
**Is your feature request related to a problem? Please describe.**

Currently Hertz does not support parseto yet

**Describe the solution you'd like**

Add parseto middleware

### PR DESCRIPTION
**Pull request by Claude-Zq**
#### What type of PR is this?
feat

#### What this PR does / why we need it (en: English/zh: Chinese):

en: add paseto middleware
zh: 添加 `paseto` 中间件

#### Which issue(s) this PR fixes:
Fixes https://github.com/cloudwego/hertz/issues/618

### CODE REVIEW

**Comment by li-jin-gou** [comment]
PATL @zstone12 @L2ncE

**Comment by li-jin-gou** [comment]
Also I'm curious if there are any practical use scenarios? verify that your middleware is usable in complex scenarios.

**Comment by Claude-Zq** [comment]
> Also I'm curious if there are any practical use scenarios? verify that your middleware is usable in complex scenarios.

soon will be used in [FreeCar](https://github.com/CyanAsterisk/FreeCar)

File: generator.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	}, nil
+}
+
+func NewV3SignFunc(asymmetricKey string, implicit []byte) (GenTokenFunc, error) {
+	key, err := paseto.NewV3AsymmetricSecretKeyFromHex(asymmetricKey)
+	if err != nil {
+		return nil, err
+	}
+	return func(stdClaims *StandardClaims, customClaims map[string]interface{}, footer []byte) (string, error) {
+		if tk, err := newTokenFromClaims(stdClaims, customClaims, footer); err != nil {
```
Comment (L2ncE) [inline generator.go:92]: using `utils.H` instead of `map[string]interface{}`

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,30 @@
+module github.com/hertz-contrib/paseto
+
+go 1.18
+
+require (
+	aidanwoods.dev/go-paseto v1.3.0
+	github.com/cloudwego/hertz v0.5.2
```
Comment (L2ncE) [inline go.mod]: update hertz version

File: parser.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	key, err := paseto.V4SymmetricKeyFromHex(symmetricPubKey)
+	if err != nil {
+		return nil, nil
+	}
+	return func(token string) (*paseto.Token, error) {
+		return parser.ParseV4Local(key, token, implicit)
+	}, nil
+}
+
+func NewV4PublicParseFunc(asymmetricPubKey string, implicit []byte, options ...ParseOption) (ParseFunc, error) {
```
Comment (L2ncE) [inline parser.go:108]: Add comments to all public functions

File: README.md [hunk 4]
```diff
@@ -1 +1,93 @@
-# .github
\ No newline at end of file
+# PASETO (This is a community driven project)
+
```
Comment (L2ncE) [inline README.md:2]: Can you introduce the difference of each version of paseto through a table

**Review by L2ncE — CHANGES_REQUESTED** [review]

File: paseto_test.go [hunk 5]
```diff
@@ -0,0 +1,698 @@
...
+	}, nil, nil)
+}
+
+func TestDefault(t *testing.T) {
+	engine := setupEngine("/paseto", New())
+	resp1 := ut.PerformRequest(engine, "POST", "/paseto", nil, ut.Header{
+		Key:   "Authorization",
+		Value: testV4PublicToken,
+	}).Result()
+	assert.DeepEqual(t, http.StatusOK, resp1.StatusCode())
```
Comment (L2ncE) [inline paseto_test.go]: using `"github.com/cloudwego/hertz/pkg/protocol/consts"` instead of `http`

File: generator.go (hunk 1 above)
Comment (Claude-Zq) [reply generator.go:92]: fixed

File: go.mod (hunk 2 above)
Comment (Claude-Zq) [reply go.mod]: fixed

File: parser.go (hunk 3 above)
Comment (Claude-Zq) [reply parser.go:108]: fixed

File: README.md (hunk 4 above)
Comment (Claude-Zq) [reply README.md:2]: fixed

File: paseto_test.go (hunk 5 above)
Comment (Claude-Zq) [reply paseto_test.go]: fixed

**Review by L2ncE — APPROVED** [review]
LGTM

**Comment by Claude-Zq** [comment]
PATL @li-jin-gou @zstone12

**Review by li-jin-gou — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**